src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -897,14 +867,7 @@ void c_typecheck_baset::typecheck_expr_symbol(exprt &expr)
 void c_typecheck_baset::typecheck_side_effect_statement_expression(
   side_effect_exprt &expr)
 {
-  if(expr.operands().size()!=1)
-  {
-    error().source_location = expr.source_location();
-    error() << "statement expression expects one operand" << eom;
-    throw 0;
-  }
-
-  codet &code=to_code(expr.op0());
+  codet &code = to_code(to_unary_expr(expr).op());
```

Comment:
Have we established before that this is a unary expression? Else this loses a valuable error message.

Reply:
That should be a guarantee given by the parser, i.e., there should not be a way for the user to trigger this.